demo-opensuse/README.md

```diff
@@ -0,0 +1,248 @@
+# │   ├── carl
+# │   └── carl.pub
+# └── owner_alice
+#     ├── alice
+#     ├── alice.pub
+#     └── create_layout.py
+```
+
+### Define software supply chain layout (Alice)
+First, we will create a new package on open build server and we will need to define the software supply chain layout. To simplify this process, we provide a script that generates a simple layout for the purpose of the demo.
+
+In this software supply chain layout, we have Alice, who is the project owner that creates the layout, Bob, who clones the project's repo and performs some pre-packaging editing (update changelog), Carl, then tests the build and verifies that its fit to ship, Carl then commits the package which triggers open build server to build the RPMs.
+
+First, signup for an account on opensuse build service at
+https://build.opensuse.org/ICSLogin/
+
+Export your username as environment variable so that you can copy paste the demo commands
+```
+username=<username>
+```
+
+Checkout your default project. You will be prompted for your username and password which is stored in `~/.oscrc` as plain text.
+```
+cd owner_alice/
+osc checkout home:$username
+```
+
+Specify the build target
+```
+osc meta prj -e home:$username
+```
+
+This will open a template xml in your favourite editor. For this demo uncomment the first one.
+```
+<repository name="openSUSE_Factory">
+   <path project="openSUSE:Factory" repository="standard" />
+   <arch>x86_64</arch>
+   <arch>i586</arch>
+ </repository>
+```
+
+Create a new package named connman in your home project
+```
+cd home:$username
+osc meta pkg -e home:$username connman
+```
+
+osc will open a template xml, fill out name, title, description.
+You can see your package here https://build.opensuse.org/package/show/home:$username/connman (replace $username in url with actual username).
+
+Update the local copy from the central server. you will get a new connman directory
+```
+osc up
+```
+
+Create and sign the software supply chain layout on behalf of Alice
+```
+cd ..
+python create_layout.py
+```
+The script will create a layout, add Bob's and Carl's public keys (fetched from
+their directories), sign it with Alice's private key and dump it to `root.layout`.
+In `root.layout`, you will find that (besides the signature and other information)
+there are four steps, `clone`, `update-changelog`, `test` and `package`, that
+the functionaries Bob and Carl, identified by their public keys, are authorized
+to perform.
+
+### Clone project source code (Bob)
+Now, we will take the role of the functionary Bob and perform the step
+`clone` on his behalf, that is we use in-toto to clone the project repo from GitHub and
+record metadata for what we do. Execute the following commands to change to Bob's
+directory and perform the step.
+
+```
+cd ../functionary_bob
+mv ../owner_alice/home:$username/ ./
+cd home:$username
+in-toto-record --step-name clone --key ../bob start
+git clone https://github.com/shikherverma/connman.git connman-src
```

Comment:
the fork is just https://build.opensuse.org/package/show/network:utilities/connman# put in a git repo. Getting it from build.opensuse.org also dowloaded `osc` metadata which we then had to remove. Using a fork makes the demo cleaner.